src/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaLengthRestriction.php

```diff
@@ -50,6 +49,6 @@ public function create(Constraint $constraint, ApiProperty $propertyMetadata): a
      */
     public function supports(Constraint $constraint, ApiProperty $propertyMetadata): bool
     {
-        return $constraint instanceof Length && null !== ($type = $propertyMetadata->getBuiltinTypes()[0] ?? null) && Type::BUILTIN_TYPE_STRING === $type->getBuiltinType();
+        return $constraint instanceof Length && null !== ($type = $propertyMetadata->getBuiltinTypes()[0] ?? null);
```

Comment:
We don't want to remove this check, but only remove it when we are in a `Collection` constraint.